kafka/integration_test.go

```diff
@@ -787,7 +621,7 @@ func xTestProducerBatchChannelDR(t *testing.T) {
 // use opaque string to locate the matching test message for message verification
 func findExpectedMessage(expected []*testmsgType, opaque string) *testmsgType {
 	for i, m := range expected {
-		if expected[i].msg.Opaque.(string) == opaque {
+		if expected[i].msg.Opaque != nil && expected[i].msg.Opaque.(string) == opaque {
```

Comment:
I think we actually want to compare the Opaque pointer value to be equal, not the contents.
This isn't critical though, so you dont need to fix it if it means a larger test overhaul.